pkg/operator/controller/controller_service_monitor.go

```diff
@@ -85,6 +88,20 @@ func (r *reconciler) currentServiceMonitor(ci *operatorv1.IngressController) (*u
 		Version: "v1",
 	})
 	if err := r.Client.Get(context.TODO(), serviceMonitorName(ci), sm); err != nil {
+		if meta.IsNoMatchError(err) {
+			// Refresh kube client with latest rest scheme/mapper.
+			kClient, err := operatorclient.NewClient(r.KubeConfig)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create kube client: %v", err)
+			}
+			r.Client = kClient
```

Comment:
If we're not going to guard, should we at least note that this is not thread safe?

Reply:
We have noted that this client field is not thread safe in the field declaration.